source/navier_stokes_matrix.cc

```diff
@@ -871,11 +871,18 @@ NavierStokesMatrix<dim>::local_operation(
                 Assert(false, ExcNotImplemented());
 #pragma GCC diagnostic push
             }
+
           // add pressure
           for (unsigned int d = 0; d < dim; ++d)
             {
               grad_u[d][d] =
                 tmu_times_2 * grad_u[d][d] + parameters.tau_grad_div * divergence;
+
+              // Subtract the volumetric part from the rate of deformation tensor.
+              if (parameters.constitutive_type ==
+                  FlowParameters::newtonian_compressible_stokes_hypothesis)
+                grad_u[d][d] -= tmu_times_2 * divergence / static_cast<double>(dim);
```

Comment:
:)

Reply:
I suppose this could be fixed within the `VectorizedArray` class ^^.